vhdbuilder/scripts/automate_generate_version.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -euxo pipefail
+
+build_ids=$1
+global_image_version=""
+for build_id in $build_ids; do
+    for artifact in $(az pipelines runs artifact list --run-id $build_id | jq -r '.[].name'); do    # Retrieve what artifacts were published
+        if [[ $artifact == *"publishing-info"* ]]; then
+            az pipelines runs artifact download --artifact-name $artifact --path $(pwd) --run-id $build_id
+            current_image_version=$(jq -r .image_version < vhd-publishing-info.json)
+            if [[ $global_image_version != $current_image_version ]]; then
+                if [[ -z $global_image_version ]]; then
+                    global_image_version=$current_image_version
+                else 
+                    exit 1
+                fi
+            fi
+        fi
```

Comment:
Will there be multiple artifacts matching the condition ever? From the code I feel there wouldn't be. 

It might be better to use to jq's filter to get the artifact you want. Also if the filter results in multiple items with the same name (maybe in the future there is a change that causes this) then just error out instead of using the wrong value.

Just put a TODO for now so that we can get it checked in.

Reply:
as discussed offline. Lets just put the context as comment.